tooling/vdu-prechecks/vdu_precheck.py

```diff
@@ -0,0 +1,302 @@
+#!/usr/bin/env python
+
+'''
+Script to check whether a server is ready to be used for a vDU, via redfish
+'''
+
+import argparse
+import pprint
+import requests
+import urllib3
+
+
+parser = argparse.ArgumentParser(
+    prog='vdu_precheck.py',
+    description='queries redfish to determine if a system is ready to run a vDU workload'
+)
+parser.add_argument("-i","--ip", help="redfish ip", required=True)
+parser.add_argument("-u","--user", help="redfish user", required=True)
+parser.add_argument("-p","--password", help="redfish password", required=True)
+parser.add_argument("-t","--type", help="vdu type (mb vs. lb)", required=True)
+parser.add_argument("-d","--debug", help="enable debugging", action="store_true")
+parser.add_argument("-v","--verify", help="ssl verify", action="store_true")
+args = parser.parse_args()
+
+if not args.verify:
+    urllib3.disable_warnings(urllib3.exceptions.InsecureRequestWarning)
+
+reference_bios = {
+    "HPE": {
+        # from partner
+        "ProcessorConfigTDPLevel": "Level2",
+        "EnhancedProcPerf": "Enabled",
+        "Sriov": "Enabled",
+        "SubNumaClustering": "Disabled",
+        "NumaGroupSizeOpt": "Flat",
+        "IntelPriorityBaseFreq": "Enabled",
+        "ProcX2Apic": "Enabled",
+        "MemPatrolScrubbing": "Disabled",
+        "ThermalConfig": "OptimalCooling",
+        # rh docs
+        "UncoreFreqScaling": "Maximum",
+        "ProcTurbo": "Enabled",
+    },
+    "Dell Inc.": {
+        # from partner
+        "ProcConfigTdp": "Level2",
+        # EnhancedProcPerf
+        "SriovGlobalEnable": "Enabled",
+        "SubNumaCluster": "Disabled",
+        # NumaGroupSizeOpt
+        # IntelPriorityBaseFreq
+        "ProcX2Apic": "Enabled",
+        "MemPatrolScrub": "Standard",
+        # ThermalConfig
+        # rh docs
+        "UncoreFrequency": "Disabled",
+        "ProcC1E": "Disabled",
+        "ProcTurboMode": "Enabled"
+    }
+}
+
+supported_nics = ['E810', 'XXV710']
+
+dpp = pprint.PrettyPrinter(indent=2)
+
+def get_url(url):
+    '''
+    retrives a URL and returns the json formatted response
+    '''
+
+    request = requests.get(url, verify=args.verify, auth=(args.user, args.password))
```

Comment:
That seems to be the only line using requests.
If so, you can use the following to remove the need for an extra dependency

```
import base64
import ssl
from urllib.request import urlopen, Request

headers = {'Content-Type': 'application/json', 'Accept': 'application/json'}
credentials = base64.b64encode(bytes(f'{user}:{password}', 'ascii')).decode('utf-8')
headers["Authorization"] = f"Basic {credentials}
context = ssl.create_default_context()
context.check_hostname = False
context.verify_mode = ssl.CERT_NONE
request = Request(url, headers=headers)
return json.loads(urlopen(request, context=context).read())
```